app/components/tables/holidays_table/HolidaysTable.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+    useReactTable,
+    flexRender,
+    getCoreRowModel,
+    getPaginationRowModel,
+    getFilteredRowModel,
+    getSortedRowModel,
+} from "@tanstack/react-table";
+import useAxiosPromise from "../../../hooks/useAxiosPromise";
+import React, { useMemo, useState, useEffect } from "react";
+import HolidaysTableHeaders from "@/components/tables/holidays_table/table_element/HolidaysTableHeaders";
+import { Holiday } from "../../../types/Holiday";
+interface HolidaysTableProps {
+    data: Holiday[];
+}
+
+
+export default function NewHolidaysTable({ data }: HolidaysTableProps): JSX.Element {
```

Comment:
The `JSX.Element` return type is not necessary here. Also to match the rest of our React components, let's define this line using `const HolidaysTable = ({ data }) => {`. Then on the last line of this file we use `export default HolidaysTable;`